packages/aft/lib/src/commands/generate/generate_amplify_swift_command.dart

```diff
@@ -0,0 +1,288 @@
+    ),
+    const PluginConfig(
+      name: 'Amplify',
+      remotePathToSource: 'Amplify',
+    ),
+  ];
+
+  /// Downloads Amplify Swift from GitHub into a temporary directory.
+  Future<Directory> _downloadRepository() => _cloneMemo.runOnce(() async {
+        final cloneDir =
+            await Directory.systemTemp.createTemp('amplify_swift_');
+        logger
+          ..info('Downloading Amplify Swift...')
+          ..verbose('Cloning repo to ${cloneDir.path}');
+        await runGit(
+          [
+            'clone',
+            // https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/
+            '--filter=tree:0',
+            'https://github.com/aws-amplify/amplify-swift.git',
+            cloneDir.path,
+          ],
+          echoOutput: verbose,
+        );
+        logger.info('Successfully cloned Amplify Swift Repo');
+        return cloneDir;
+      });
+
+  /// Checks out [ref] in [pluginDir].
+  Future<Directory> _checkoutRepositoryRef(
+    Directory pluginDir,
+    String ref,
+  ) async {
+    logger
+      ..info('Checking out target branch: $ref')
+      ..verbose('Creating git work tree in $pluginDir');
+    final worktreeDir =
+        await Directory.systemTemp.createTemp('amplify_swift_worktree_');
+    try {
+      await runGit(
+        ['worktree', 'add', worktreeDir.path, ref],
+        processWorkingDir: pluginDir.path,
+      );
+    } on Exception catch (e) {
+      if (e.toString().contains('already checked out')) {
+        return pluginDir;
+      }
+      rethrow;
+    }
+    return worktreeDir;
+  }
+
+  /// Sets up the Amplify Swift repo for use later
+  Future<void> _setupRepo() async {
+    if (_repoCache[branchTarget] != null) {
+      return;
+    }
+    Directory repoDir;
+    repoDir = await _downloadRepository();
+    repoDir = await _checkoutRepositoryRef(repoDir, branchTarget);
+
+    _repoCache[branchTarget] = repoDir;
+  }
+
+  /// Returns the directory for the plugin at [path].
+  Future<Directory> _pluginDirForPath(String path) async {
+    final repoDir = _repoCache[branchTarget];
+    if (repoDir == null) {
+      exitError('No cached repo for branch $branchTarget');
+    }
+
+    return Directory.fromUri(
+      repoDir.uri.resolve(path),
+    );
+  }
+
+  /// Generates the Amplify Swift plugin for [plugin].
+  Future<void> _generatePlugin(PluginConfig plugin) async {
+    logger.info('Selecting source files for ${plugin.name}...');
+
+    final repoDir = await _pluginDirForPath(plugin.remotePathToSource);
+
+    final outputDir = Directory(p.join(outputPath, plugin.name));
+
+    // Clear out the directory if it already exists.
+    // This is to ensure that we don't have any stale files.
+    if (await outputDir.exists()) {
+      logger.verbose(
+        'Clearing out existing plugin directory for ${plugin.name}...',
+      );
+      await _setPermissions(outputDir.path, PermissionScope.readExecuteWrite);
```

Comment:
Q: Do you need to set permissions to execute if you are just removing the files?

Reply:
Q: What permissions exist without this?

Reply:
After looking into this more, turns out git doesn't preserve file permissions. Will remove this logic.